referenceframe/frame_system.go

```diff
@@ -308,6 +309,55 @@ func (sfs *simpleFrameSystem) FrameSystemSubset(newRoot Frame) (FrameSystem, err
 	return newFS, nil
 }
 
+// FrameSystemToPCD in a framesystem and returns a map where all elements are the point representation
+// of their geometry type with respect to the world.
+func FrameSystemToPCD(system FrameSystem, inputs map[string][]Input) (map[string][]r3.Vector, error) {
+	vectorMap := make(map[string][]r3.Vector)
+	for _, name := range system.FrameNames() {
+		currentFrame := system.Frame(name)
+		parent, err := system.Parent(currentFrame)
+		if err != nil {
+			return nil, err
+		}
+
+		geosInFrame, err := currentFrame.Geometries(inputs[name])
```

Comment:
We can't guarantee the contents of `inputs` and whether information for any particular frame is in there.

We should check whether `name` is in `inputs`, and if it's not, then let's pass `make([]Input, len(currentFrame.DoF())`